### PR TITLE
Add ddvk-hacks

### PR DESCRIPTION
**Pull request by matteodelabre**
Patches Xochitl on install, restores original binary on remove. Tested on rM1, version 2.5.0.27.

### CODE REVIEW

**Comment by raisjn** [comment]
awesome! this day has been a long time coming :-D

cc @ddvk

File: package/ddvk-hacks/package [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    echo
+    echo "Device: $device"
+    echo "Xochitl version: $xochitl_version"
+    echo "Patch version: $patch_version"
+    echo
+
+    if ! sha1sum -c <(echo "$original_hash  $_xochitl_path") > /dev/null 2>&1; then
+        echo "Invalid Xochitl checksum"
+        echo "Maybe ddvk-hacks are already installed?"
```
Comment (Eeems) [inline package/ddvk-hacks/package]: Perhaps check the backup location that xochitl gets placed in when using @ddvk's script?

Comment (matteodelabre) [reply package/ddvk-hacks/package]: Good point. If we detect its presence, should we just warn the user and ask them to do the cleanup? Or use the upstream script’s backup to restore Xochitl, then re-run our script as normal?

Comment (Eeems) [reply package/ddvk-hacks/package]: Not quite sure. @ddvk what are you comfortable with?

**Comment by LinusCDE** [comment]
Works great on my rM 2! Best tag me in discord once the details are fleshed out. I can re-review (and most likely approve) it then.

**Review by Eeems — DISMISSED** [review]
Works fine for me on my rM1

**Comment by Eeems** [comment]
Just waiting on input from @ddvk before approving.

File: package/ddvk-hacks/package (hunk 1 above)
Comment (ddvk) [reply package/ddvk-hacks/package]: probably it is better to let the user do the cleanup

**Comment by ddvk** [comment]
> Just waiting on input from @ddvk before approving.

what happens after the device has been upgraded? 
the way i see it, the package has to be foce reinstalled

**Comment by LinusCDE** [comment]
To loosely repeat some of the discussions in discord relating to this ([starting here])(https://discordapp.com/channels/385916768696139794/756853330382422037/804318381644972033)):

The package doesn't contain any files outside of `/opt` and `/home/root` so I won't get automatically reinstalled after an upgrade. This behaviour also may not be fully wished or has to be altered since right after an update, there will most likely not be a patch for the new xochitl.

There was also some discussion on how to treat and where to store the backup of the xochitl binary do we don't accidentally downgrade it.

**Comment by Eeems** [comment]
So what is this waiting on?

**Comment by matteodelabre** [comment]
I don’t see any remaining issue with the PR. So this is just waiting on someone to review it.

**Review by Eeems — APPROVED** [review]

**Pull request merged**